tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/macro/MacroGeneratorIntegrationTest.xtend

```diff
@@ -0,0 +1,282 @@
+package org.testeditor.tcl.dsl.jvmmodel.macro
+
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.dsl.common.testing.DummyFixture
+
+class MacroGeneratorIntegrationTest extends org.testeditor.tcl.dsl.jvmmodel.AbstractTclGeneratorIntegrationTest {
+
+	@Before
+	def void setup() {
+		parseAml(DummyFixture.amlModel)
+		parseTcl('''
+			package com.example
+			
+			# MyMacroCollection
+			
+			## EmptyMacro
+				template = "Do nothing"
+			
+			## EmptyNestedMacro
+				template = "Do nothing nested"
+				Macro: MyMacroCollection
+				- Do nothing
+			
+			## EmptyMacroWithUnusedParameter
+				template = "Do nothing with" ${unused}
+			
+			## ReadMacro
+				template = "Read some values"
+				Component: GreetingApplication
+				- value = Read value from <bar>
```

Comment:
I see that this is used by a test to check whether repeated invocation will be a problem ... but ... this should actually be warned/erred of by the validator. Since values cannot be returned and this value is not used within this macro